test/iht/utils/PropertyAndMortgageHelperTest.scala

```diff
@@ -0,0 +1,194 @@
+      allAssets = Some(CommonBuilder.buildAllAssetsWithAllSectionsFilled),
+      allLiabilities = Some(CommonBuilder.buildAllLiabilitiesWithAllSectionsFilled))
+
+    "updatePropertyList is called" must {
+
+      "return an empty list if there are no properties owned" in {
+        val result = helper.updatePropertyList(propertiesNo, appDetailsWithoutProperties)
+        result shouldBe Nil
+      }
+
+      "return a list of existing properties if there are properties owned" in {
+        val appDetails = appDetailsWithProperties.copy(propertyList = propertyList)
+        val result = helper.updatePropertyList(propertiesYes, appDetails)
+        result shouldBe propertyList
+      }
+
+    }
+
+    "isMortgagesLargerThanProperties is called" must {
+
+      "return true if there are more mortgages than properties" in {
+        val result = helper.isMortgagesLargerThanProperties(appDetailsWithoutProperties.copy(
+          allLiabilities = Some(CommonBuilder.buildAllLiabilitiesWithAllSectionsFilled)))
+        result shouldBe true
+      }
+
+      "return false if there are not more mortgages than properties" in {
+        val result = helper.isMortgagesLargerThanProperties(appDetailsWithProperties.copy(propertyList = propertyList))
+        result shouldBe false
+      }
+
+    }
+
+    "reduceMortgagesToMatchProperties is called" must {
+
+      "return 0 mortgages when there are 0 properties" in {
+        val result = helper.reduceMortgagesToMatchProperties(appDetailsWithoutProperties.copy(
+          allLiabilities = Some(CommonBuilder.buildAllLiabilitiesWithAllSectionsFilled)))
+        result shouldBe Nil
+      }
+
+      "return 1 mortgage when there is 1 property and mortgage details have been entered for it" in {
+        val result = helper.reduceMortgagesToMatchProperties(appDetailsWithoutProperties.copy(
+          propertyList = List(CommonBuilder.property),
+          allLiabilities = Some(CommonBuilder.buildAllLiabilitiesWithAllSectionsFilled)))
+        result.size shouldBe 1
+      }
+
+      "return 2 mortgages when there are 2 properties and mortgage details have been entered for 2 properties" in {
+        val result = helper.reduceMortgagesToMatchProperties(appDetailsWithoutProperties.copy(
+          propertyList = List(CommonBuilder.property, CommonBuilder.property2),
+          allLiabilities = Some(CommonBuilder.buildAllLiabilitiesWithAllSectionsFilled)))
+        result.size shouldBe 2
+      }
+
+      "return 2 mortgages when there are 3 properties and mortgage details have only been entered for 2 properties" in {
+        val result = helper.reduceMortgagesToMatchProperties(appDetailsWithoutProperties.copy(
+          propertyList = List(CommonBuilder.property,
+            CommonBuilder.property2,
+            CommonBuilder.property2.copy(id = Some("3"))),
+          allLiabilities = Some(CommonBuilder.buildAllLiabilitiesWithAllSectionsFilled)))
+        result.size shouldBe 2
```

Comment:
There is already one there. I'll change the name of the test to make it clearer.